File: java/src/main/java/com/hindu/calendar/model/SolarCalendarType.java
package com.hindu.calendar.model;

public enum SolarCalendarType {
    TAMIL(1, 78, 79, "Saka", new String[]{
            "", "Chithirai", "Vaikaasi", "Aani", "Aadi", "Aavani", "Purattaasi",
            "Aippasi", "Karthikai", "Maargazhi", "Thai", "Maasi", "Panguni"
    }),
    BENGALI(1, 593, 594, "Bangabda", new String[]{
            "", "Boishakh", "Joishtho", "Asharh", "Srabon", "Bhadro", "Ashshin",
            "Kartik", "Ogrohaeon", "Poush", "Magh", "Falgun", "Choitro"
    }),
    ODIA(1, 78, 79, "Saka", new String[]{
            "", "Baisakha", "Jyeshtha", "Ashadha", "Shravana", "Bhadrapada", "Ashvina",
            "Kartika", "Margashirsha", "Pausha", "Magha", "Phalguna", "Chaitra"
    }),
    MALAYALAM(5, 824, 825, "Kollam", new String[]{
            "", "Chingam", "Kanni", "Thulam", "Vrishchikam", "Dhanu", "Makaram",
            "Kumbham", "Meenam", "Medam", "Edavam", "Mithunam", "Karkadakam"
    });

    private final int firstRashi;
    private final int gyOffsetOn;
    private final int gyOffsetBefore;
    private final String eraName;
    private final String[] months;

    SolarCalendarType(int firstRashi, int gyOffsetOn, int gyOffsetBefore,
                      String eraName, String[] months) {
        this.firstRashi = firstRashi;
        this.gyOffsetOn = gyOffsetOn;
        this.gyOffsetBefore = gyOffsetBefore;
        this.eraName = eraName;
        this.months = months;
    }

    public int firstRashi() { return firstRashi; }
    public int gyOffsetOn() { return gyOffsetOn; }
    public int gyOffsetBefore() { return gyOffsetBefore; }
    public String eraName() { return eraName; }

    public String monthName(int month) {
        if (month < 1 || month > 12) return "???";
        return months[month];
    }

    public static SolarCalendarType fromString(String name) {
        return switch (name.toLowerCase()) {
            case "tamil" -> TAMIL;
            case "bengali" -> BENGALI;
            case "odia" -> ODIA;
            case "malayalam" -> MALAYALAM;
            default -> throw new IllegalArgumentException("Unknown solar calendar: " + name);
        };
    }
}

File: swift/Sources/HinduCalendar/Model/SolarCalendarType.swift
public enum SolarCalendarType {
    case tamil
    case bengali
    case odia
    case malayalam

    public var firstRashi: Int {
        switch self {
        case .tamil: return 1
        case .bengali: return 1
        case .odia: return 1
        case .malayalam: return 5
        }
    }

    public var yearStartRashi: Int {
        switch self {
        case .tamil: return 1
        case .bengali: return 1
        case .odia: return 6
        case .malayalam: return 5
        }
    }

    public var gyOffsetOn: Int {
        switch self {
        case .tamil: return 78
        case .bengali: return 593
        case .odia: return 592
        case .malayalam: return 824
        }
    }

    public var gyOffsetBefore: Int {
        switch self {
        case .tamil: return 79
        case .bengali: return 594
        case .odia: return 593
        case .malayalam: return 825
        }
    }

    public var eraName: String {
        switch self {
        case .tamil: return "Saka"
        case .bengali: return "Bangabda"
        case .odia: return "Amli"
        case .malayalam: return "Kollam"
        }
    }

    private static let tamilMonths = [
        "", "Chithirai", "Vaikaasi", "Aani", "Aadi", "Aavani", "Purattaasi",
        "Aippasi", "Karthikai", "Maargazhi", "Thai", "Maasi", "Panguni"
    ]

    private static let bengaliMonths = [
        "", "Boishakh", "Joishtho", "Asharh", "Srabon", "Bhadro", "Ashshin",
        "Kartik", "Ogrohaeon", "Poush", "Magh", "Falgun", "Choitro"
    ]

    private static let odiaMonths = [
        "", "Baisakha", "Jyeshtha", "Ashadha", "Shravana", "Bhadrapada", "Ashvina",
        "Kartika", "Margashirsha", "Pausha", "Magha", "Phalguna", "Chaitra"
    ]

    private static let malayalamMonths = [
        "", "Chingam", "Kanni", "Thulam", "Vrishchikam", "Dhanu", "Makaram",
        "Kumbham", "Meenam", "Medam", "Edavam", "Mithunam", "Karkadakam"
    ]

    public func monthName(_ month: Int) -> String {
        guard month >= 1, month <= 12 else { return "???" }
        switch self {
        case .tamil: return Self.tamilMonths[month]
        case .bengali: return Self.bengaliMonths[month]
        case .odia: return Self.odiaMonths[month]
        case .malayalam: return Self.malayalamMonths[month]
        }
    }

    public static func fromString(_ name: String) -> SolarCalendarType? {
        switch name.lowercased() {
        case "tamil": return .tamil
        case "bengali": return .bengali
        case "odia": return .odia
        case "malayalam": return .malayalam
        default: return nil
        }
    }
}
